lib/tasks/sw.rake

```diff
@@ -60,6 +58,38 @@ namespace :sw do
     harvester.harvest_from_wos_id_cap_profile_id_report args[:path_to_report]
   end
 
+  desc 'Compare publications (IDs) returned by APIs for an Author identified by their name [last,first,middle]'
+  task :wos_publications_for_name, [:last, :first, :middle] => :environment do |_t, args|
+    fail "last name argument is required" unless args[:last].present?
+    fail "first name argument is required" unless args[:first].present?
+    author_name = ScienceWire::AuthorName.new(args[:last], args[:first], args[:middle])
+    attribs = ScienceWire::AuthorAttributes.new(author_name, '', [], ScienceWireHarvester.new.default_institution)
+    puts "Querying ScienceWire for #{author_name.inspect}"
+    ids = ScienceWire::HarvestBroker.new(nil, ScienceWireHarvester.new).ids_from_dumb_query(attribs)
+    puts "\n" << ids.sort if ids.count > 0
+    puts "\nQuerying WebOfScience for #{author_name.full_name}"
+    records = WosQueries.new(WosClient.new(Settings.WOS.AUTH_CODE)).search_by_name(author_name.full_name)
+    uids = records.uids.sort
+    puts uids
+    puts "\n#{ids.count} ScienceWire IDs"
+    puts "#{uids.count} WebOfScience IDs"
+    wos_ids     = uids.select { |id| id =~ /^WOS:/ }.map { |id| id[4..-1] } # match and strip prefix
+    medline_ids = uids.select { |id| id =~ /^MEDLINE:/ }.map { |id| id[8..-1] } # match and strip prefix
```

Comment:
The select and regex is working here, no problem, just wondering if it could be a Regex.match() but I don't care.  It might be more efficient to instantiate the Regex once, but I dunno for sure.  e.g.
```ruby
> id = 'WOS:1234'
=> "WOS:1234"
> wos_regex = Regexp.new(/^WOS:(.*)/)
=> /^WOS:(.*)/
> wos_regex.match(id)
=> #<MatchData "WOS:1234" 1:"1234">
> wos_regex.match(id)[1]
=> "1234"
```
It might be used like:
```ruby
ids.map {|id| match = wos_regex.match(id); match[1] if match }.compact
```